homeassistant/components/p1_monitor/coordinator.py

```diff
@@ -60,7 +60,7 @@ def __init__(
 
         self.p1monitor = P1Monitor(
             host=self.config_entry.data[CONF_HOST],
-            port=self.config_entry.data[CONF_PORT],
+            port=int(self.config_entry.data[CONF_PORT]),
```

Comment:
Before it was changed to a numberSelector it worked fine 🤷🏻‍♂️ 

Could apply the same as in the example, but would a migration still be necessary?

Reply:
Have you been able to confirm via the diagnostics what is currently stored in the entry data?
I that is wrong, then I think a minor version migration would be best.

Reply:
I've investigated this further.
Since the failure occurs within the config flow, there is no need to do a migration as the config entry is never created.

You can just amend the config flow schema and it should be fine...